### PR TITLE
feat: add custom system prompt management CLI

### PR DESCRIPTION
**Pull request by sogoiii**
## Purpose
  Add comprehensive CLI commands for managing custom system prompts, enabling users to override the default prompt with their own custom instructions for commit message generation.

  ## Summary
  - ✅ New `gac prompt` subcommand suite with show/set/clear operations
  - ✅ Interactive prompt editor with `--edit` flag for terminal-based editing
  - ✅ File-based prompt import with `--file` option
  - ✅ Environment variable support via `GAC_SYSTEM_PROMPT_PATH`
  - ✅ 100% test coverage for all commands and edge cases

  ## Changes

  ### CLI Commands
  **`gac prompt show`** - Display the currently active system prompt (custom or default)

  **`gac prompt set --edit`** - Open interactive terminal editor with:
  - Multi-line text editing via prompt_toolkit
  - Terminal size adaptation
  - Save/cancel shortcuts with visual hints
  - Pre-populated with existing prompt content

  **`gac prompt set --file <path>`** - Import custom prompt from file

  **`gac prompt clear`** - Remove custom prompt and revert to default

  ### Prompt Precedence
  1. `GAC_SYSTEM_PROMPT_PATH` environment variable (highest priority)
  2. `~/.config/gac/custom_system_prompt.txt` (stored configuration)
  3. Built-in default prompt (fallback)

  ### Code Structure
  - `src/gac/prompt_cli.py` (265 lines) - Main prompt management logic
  - Updated `src/gac/cli.py` - Integrated new subcommand
  - `tests/test_prompt_cli.py` (265 lines) - Comprehensive test suite

  ## Impact Analysis
  - **Scope**: CLI command addition, no breaking changes
  - **Test Coverage**: 265 lines of tests for all commands and edge cases
  - **Bundle Size**: Minimal - uses existing dependencies (prompt_toolkit)
  - **Dependencies**: None new required

  ## Validation
  - ✅ Unit Tests: Comprehensive coverage for all commands
  - ✅ Edge Cases: Handles missing files, invalid paths, empty prompts
  - ✅ Error Handling: Clear messages for all failure scenarios
  - ✅ Integration: Seamlessly integrates with existing CLI structure

## Summary by CodeRabbit

* **New Features**
  * Added a new `prompt` command group for managing system prompts
  * `prompt show`: displays the active prompt and its source
  * `prompt set`: create or update prompts via interactive editor or external file
  * `prompt clear`: remove custom prompt settings
  * Environment-variable support for specifying prompt files

* **Tests**
  * Added comprehensive tests covering show/set/clear behavior, precedence, editor flows, and error cases

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by sogoiii** [comment]
@cellwebb ok, my branch has drifted since i made the PR a week ago. I have to rebase against main.

File: src/gac/prompt_cli.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+
+@prompt.command()
+def clear() -> None:
+    """Clear custom system prompt (revert to default)."""
+    if CUSTOM_PROMPT_FILE.exists():
+        CUSTOM_PROMPT_FILE.unlink()
+        console.print(f"Custom prompt deleted: {CUSTOM_PROMPT_FILE}")
+    else:
+        console.print("No custom prompt file to delete.")
```
[the remark this answers is not in the document]
Comment (sogoiii) [reply src/gac/prompt_cli.py]: again the `GAC_SYSTEM_PROMPT_PATH` takes precedent. The env var is the original way of handling custom prompts. The new command is mean to handle new users who wont use the env var.
